docproc/src/main/java/com/yahoo/docproc/jdisc/DocumentProcessingHandler.java

```diff
@@ -93,7 +93,7 @@ public DocumentProcessingHandler(ComponentRegistry<DocprocService> docprocServic
                                            (params.getMaxQueueTimeMs() > 0)
                                                ? new ThroughputLimitQueue<>(params.getMaxQueueTimeMs())
                                                : (params.getMaxQueueTimeMs() < 0)
-                                                   ? new LinkedBlockingQueue<>()
```

Comment:
I don't think we want this to block, so you should probably do
(maxNumThreads > 0) ? new LinkedBlockingQueue<>() : new SynchronousQueue<>()
?

Reply:
Good point!